eventually/customprofile/views.py

```diff
@@ -1,10 +1,121 @@
 
 
 class CustomProfileView(View):
-    """Team view handles GET, POST, PUT, DELETE requests."""
+    """
+    CustomProfile view that handles GET, POST, PUT, DELETE requests and provides appropriate
+    operations with profile model.
+    """
+
+    def get(self, request, user_id):
+        """
+        Method that handles POST request.
+
+        :param request: the accepted HTTP request.
+        :type request: `HttpRequest object`
+
+        :param user_id: ID of the certain user.
+        :type user_id: `int`
+
+        :return: the response with certain profile information when the profile was successfully
+                 created or response with 400 or 404 failed status code.
+
+        :rtype: `HttpResponse object`.
+        """
+        if user_id:
+            user = CustomUser.get_by_id(user_id)
+            if not user:
+                return HttpResponse(status=404)
+
+        profile = user.customprofile
+        if profile:
+            profile = profile.to_dict()
+            user = user.to_dict()
+            information = user.copy()
+            information.update(profile)
+            return JsonResponse(information, status=200)
+        return HttpResponse(status=404)
+
+    def post(self, request):
+        """
+        Method that handles POST request.
+
+        :param request: the accepted HTTP request.
+        :type request: `HttpRequest object`
+
+        :return: the response with certain profile information when the profile was successfully
+                 created or response with 400 or 404 failed status code.
+
+        :rtype: `HttpResponse object.
+        """
+        user = request.user
+        data = request.body
+        if not profile_data_validator(data):
+            return HttpResponse(status=400)
+
+        requred_keys = ["user"]
+        if not required_keys_validator(data, requred_keys):
+            return HttpResponse(status=400)
+
+        data = {'user': user,
+                'hobby': data.get('hobby'),
+                'photo': data.get('photo'),
+                'birthday': datetime.date.fromtimestamp(data.get('birthday'))}
+
+        profile = CustomProfile.create(**data)
+        profile = profile.to_dict()
+        return JsonResponse(profile, status=201)
+
+    def put(self, request):
+        """
+        Method that handles PUT request.
+
+        :param request: the accepted HTTP request.
+        :type request: `HttpRequest object`
+
+        :return: response with status code 204 when profile was successfully updated
+                 or response with 400, 403 or 404 failed status code.
+
+        :rtype: `HttpResponse object."""
+
+        user = request.user
+        data = request.body
+        if not profile_data_validator(data):
+            return HttpResponse(status=400)
+
+        data = {'hobby': data.get('hobby'),
+                'photo': data.get('photo'),
+                'birthday': datetime.date.fromtimestamp(data.get('birthday'))}
```

Comment:
what will be if the `data.get('birthday')` returns `None`?

Reply:
changed